NitroxServer-Subnautica/Serialization/ResourceAssets/MonoBehaviourParsers/ReefbackSlotsDataParser.cs

```diff
@@ -0,0 +1,156 @@
+                        rotation = plant.rotation,
+                        position = plant.position,
+                        scale = plant.scale
+                    });
+                }
+            }
+
+            uint reefbackCreatureSize = resourcesFile.reader.ReadUInt32();
+            for (int i = 0; i < reefbackCreatureSize; i++)
+            {
+
+                resourcesFile.reader.ReadCountStringInt32(); // Name
+                resourcesFile.reader.Align();
+                uint fileId = (uint)resourcesFile.reader.ReadInt32();
+                ulong pathId = (ulong)resourcesFile.reader.ReadInt64(); // PathId
+                ReefbackSpawnData.ReefbackCreature creaturePrefab = new ReefbackSpawnData.ReefbackCreature();
+                AssetFileInfoEx asset = ResourceAssetsHelper.FindComponentOfType<PrefabIdentifier>(instance, pathId);
+                ulong prevPos = resourcesFile.reader.Position;
+                resourcesFile.reader.Position = asset.absoluteFilePos + 28;
+
+                resourcesFile.reader.ReadCountStringInt32(); // Empty name
+                string classId = resourcesFile.reader.ReadCountStringInt32(); // ClassId
+
+                resourcesFile.reader.Position = prevPos;
+                asset = ResourceAssetsHelper.FindTransform(instance, pathId);
+                prevPos = resourcesFile.reader.Position;
+                resourcesFile.reader.Position = asset.absoluteFilePos + 12;
+
+                Quaternion rotation = new Quaternion(
+                    resourcesFile.reader.ReadSingle(), // Quaternion X
+                    resourcesFile.reader.ReadSingle(), // Quaternion Y
+                    resourcesFile.reader.ReadSingle(), // Quaternion Z
+                    resourcesFile.reader.ReadSingle()); // Quaternion W
+
+                Vector3 position = new Vector3(
+                    resourcesFile.reader.ReadSingle(), // Position X
+                    resourcesFile.reader.ReadSingle(), // Position Y
+                    resourcesFile.reader.ReadSingle()); // Position Z
+
+                Vector3 scale = new Vector3(
+                    resourcesFile.reader.ReadSingle(), // Scale X
+                    resourcesFile.reader.ReadSingle(), // Scale Y
+                    resourcesFile.reader.ReadSingle()); // Scale Z
+
+                creaturePrefab = new ReefbackSpawnData.ReefbackCreature
+                {
+                    classId = classId,
+                    position = position,
+                    scale = scale,
+                    rotation = rotation,
+                };
+
+                resourcesFile.reader.Position = prevPos;
+
+
+                int minNumber = resourcesFile.reader.ReadInt32();
+                int maxNumber = resourcesFile.reader.ReadInt32();
+                float probability = resourcesFile.reader.ReadSingle();
+
+                ReefbackSpawnData.SpawnableCreatures.Add(new ReefbackSpawnData.ReefbackCreature()
```

Comment:
If you pass these (`ReefbackSpawnData`) in via the constructor you won't have to statically assign these.

Reply:
Considering the class was static and the places that make use of it converting the class into a constructable class and passing it around via Autofac or otherwise wouldve been more of a hassle than its worth...